### PR TITLE
firewalld: add zone target set

### PR DESCRIPTION
**Pull request by maxamillion**
##### SUMMARY

Add Zone target setting

Fixes https://github.com/ansible/ansible/issues/49232

Signed-off-by: Adam Miller <admiller@redhat.com>

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

firewalld

### CODE REVIEW

File: plugins/modules/system/firewalld.py [hunk 1]
```diff
@@ -83,6 +83,12 @@
     description:
       - Whether to run this module even when firewalld is offline.
     type: bool
+  target:
+    description:
+      - firewalld Zone target
+      - If state is set to C(absent), this will reset the target to default
+    choices: [ default, ACCEPT, DROP, REJECT ]
+    type: str
```
Comment (felixfontein) [inline plugins/modules/system/firewalld.py:91]:
```suggestion
    type: str
    version_added: 0.2.0
```

File: changelogs/fragments/firewalld_zone_target.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - firewalld - new feature, can now set ``target`` for a ``zone``
```
Comment (felixfontein) [inline changelogs/fragments/firewalld_zone_target.yml:2]:
```suggestion
  - firewalld - new feature, can now set ``target`` for a ``zone`` (https://github.com/ansible-collections/community.general/pull/526).
```

**Comment by maxamillion** [comment]
> Did you run the tests manually? Right now, they are still disabled in CI.

Yes, but the test is failing and I don't know why ... I'm going to mark the PR as `[WIP]` while I figure it out.

**Comment by maxamillion** [comment]
@felixfontein alright, got it fixed and ran integration tests successfully against RHEL 8.1

```
$(ansible)  ansible-test integration --remote rhel/8.1 firewalld --allow-disabled
Red Hat Enterprise Linux release 8.1 (Ootpa)
Running firewalld integration test role
[WARNING]: running playbook inside collection community.general

PLAY [testhost] ****************************************************************************************************************************

TASK [Gathering Facts] *********************************************************************************************************************
ok: [testhost]

TASK [setup_pkg_mgr : set_fact] ************************************************************************************************************
skipping: [testhost]

TASK [setup_pkg_mgr : set_fact] ************************************************************************************************************
skipping: [testhost]

TASK [Ensure firewalld is installed] *******************************************************************************************************
ok: [testhost]

TASK [Check to make sure the firewalld python module is available.] ************************************************************************
changed: [testhost]

TASK [start firewalld] *********************************************************************************************************************
changed: [testhost]

TASK [Ensure /run/firewalld exists] ********************************************************************************************************
ok: [testhost]

TASK [firewalld : include_tasks] ***********************************************************************************************************
included: /root/ansible_collections/community/general/tests/output/.tmp/integration/firewalld-hpod3w84-ÅÑŚÌβŁÈ/tests/integration/targets/firewalld/tasks/service_test_cases.yml for testhost

TASK [firewalld service test permanent enabled] ********************************************************************************************
changed: [testhost]

TASK [assert firewalld service test permanent enabled worked] ******************************************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld service test permanent enabled rerun (verify not changed)] *****************************************************************
ok: [testhost]

TASK [assert firewalld service test permanent enabled rerun worked (verify not changed)] ***************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld service test permanent disabled] *******************************************************************************************
changed: [testhost]

TASK [assert firewalld service test permanent disabled worked] *****************************************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld service test permanent disabled rerun (verify not changed)] ****************************************************************
ok: [testhost]

TASK [assert firewalld service test permanent disabled rerun worked (verify not changed)] **************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld : include_tasks] ***********************************************************************************************************
included: /root/ansible_collections/community/general/tests/output/.tmp/integration/firewalld-hpod3w84-ÅÑŚÌβŁÈ/tests/integration/targets/firewalld/tasks/port_test_cases.yml for testhost

TASK [firewalld port test permanent enabled] ***********************************************************************************************
changed: [testhost]

TASK [assert firewalld port test permanent enabled worked] *********************************************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld port test permanent enabled rerun (verify not changed)] ********************************************************************
ok: [testhost]

TASK [assert firewalld port test permanent enabled rerun worked (verify not changed)] ******************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld port test permanent disabled] **********************************************************************************************
changed: [testhost]

TASK [assert firewalld port test permanent disabled worked] ********************************************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld port test permanent disabled rerun (verify not changed)] *******************************************************************
ok: [testhost]

TASK [assert firewalld port test permanent disabled rerun worked (verify not changed)] *****************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld source test permanent enabled] *********************************************************************************************
changed: [testhost]

TASK [assert firewalld source test permanent enabled worked] *******************************************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld source test permanent enabled rerun (verify not changed)] ******************************************************************
ok: [testhost]

TASK [assert firewalld source test permanent enabled rerun worked (verify not changed)] ****************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld source test permanent disabled] ********************************************************************************************
changed: [testhost]

TASK [assert firewalld source test permanent disabled worked] ******************************************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld source test permanent disabled rerun (verify not changed)] *****************************************************************
ok: [testhost]

TASK [assert firewalld source test permanent disabled rerun worked (verify not changed)] ***************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld source test permanent enabled is exclusive (verify exclusive error)] *******************************************************
fatal: [testhost]: FAILED! => {"changed": false, "msg": "can only operate on port, service, rich_rule, masquerade, icmp_block, icmp_block_inversion, interface or source at once"}
...ignoring

TASK [assert firewalld source test permanent enabled is exclusive (verify exclusive error)] ************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld dmz zone target DROP] ******************************************************************************************************
changed: [testhost]

TASK [assert firewalld dmz zone target DROP present worked] ********************************************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld dmz zone target DROP rerun (verify not changed)] ***************************************************************************
ok: [testhost]

TASK [assert firewalld dmz zone target DROP present worked (verify not changed)] ***********************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld dmz zone target DROP absent] ***********************************************************************************************
changed: [testhost]

TASK [assert firewalld dmz zone target DROP absent worked] *********************************************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld dmz zone target DROP rerun (verify not changed)] ***************************************************************************
ok: [testhost]

TASK [assert firewalld dmz zone target DROP present worked (verify not changed)] ***********************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [stop firewalld] **********************************************************************************************************************
changed: [testhost]

TASK [Ensure /run/firewalld exists] ********************************************************************************************************
ok: [testhost]

TASK [firewalld : include_tasks] ***********************************************************************************************************
included: /root/ansible_collections/community/general/tests/output/.tmp/integration/firewalld-hpod3w84-ÅÑŚÌβŁÈ/tests/integration/targets/firewalld/tasks/service_test_cases.yml for testhost

TASK [firewalld service test permanent enabled] ********************************************************************************************
changed: [testhost]

TASK [assert firewalld service test permanent enabled worked] ******************************************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld service test permanent enabled rerun (verify not changed)] *****************************************************************
ok: [testhost]

TASK [assert firewalld service test permanent enabled rerun worked (verify not changed)] ***************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld service test permanent disabled] *******************************************************************************************
changed: [testhost]

TASK [assert firewalld service test permanent disabled worked] *****************************************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld service test permanent disabled rerun (verify not changed)] ****************************************************************
ok: [testhost]

TASK [assert firewalld service test permanent disabled rerun worked (verify not changed)] **************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld : include_tasks] ***********************************************************************************************************
included: /root/ansible_collections/community/general/tests/output/.tmp/integration/firewalld-hpod3w84-ÅÑŚÌβŁÈ/tests/integration/targets/firewalld/tasks/port_test_cases.yml for testhost

TASK [firewalld port test permanent enabled] ***********************************************************************************************
changed: [testhost]

TASK [assert firewalld port test permanent enabled worked] *********************************************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld port test permanent enabled rerun (verify not changed)] ********************************************************************
ok: [testhost]

TASK [assert firewalld port test permanent enabled rerun worked (verify not changed)] ******************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld port test permanent disabled] **********************************************************************************************
changed: [testhost]

TASK [assert firewalld port test permanent disabled worked] ********************************************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld port test permanent disabled rerun (verify not changed)] *******************************************************************
ok: [testhost]

TASK [assert firewalld port test permanent disabled rerun worked (verify not changed)] *****************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld source test permanent enabled] *********************************************************************************************
changed: [testhost]

TASK [assert firewalld source test permanent enabled worked] *******************************************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld source test permanent enabled rerun (verify not changed)] ******************************************************************
ok: [testhost]

TASK [assert firewalld source test permanent enabled rerun worked (verify not changed)] ****************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld source test permanent disabled] ********************************************************************************************
changed: [testhost]

TASK [assert firewalld source test permanent disabled worked] ******************************************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld source test permanent disabled rerun (verify not changed)] *****************************************************************
ok: [testhost]

TASK [assert firewalld source test permanent disabled rerun worked (verify not changed)] ***************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld source test permanent enabled is exclusive (verify exclusive error)] *******************************************************
fatal: [testhost]: FAILED! => {"changed": false, "msg": "can only operate on port, service, rich_rule, masquerade, icmp_block, icmp_block_inversion, interface or source at once"}
...ignoring

TASK [assert firewalld source test permanent enabled is exclusive (verify exclusive error)] ************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld dmz zone target DROP] ******************************************************************************************************
changed: [testhost]

TASK [assert firewalld dmz zone target DROP present worked] ********************************************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld dmz zone target DROP rerun (verify not changed)] ***************************************************************************
ok: [testhost]

TASK [assert firewalld dmz zone target DROP present worked (verify not changed)] ***********************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld dmz zone target DROP absent] ***********************************************************************************************
changed: [testhost]

TASK [assert firewalld dmz zone target DROP absent worked] *********************************************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [firewalld dmz zone target DROP rerun (verify not changed)] ***************************************************************************
ok: [testhost]

TASK [assert firewalld dmz zone target DROP present worked (verify not changed)] ***********************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

PLAY RECAP *********************************************************************************************************************************
testhost                   : ok=79   changed=19   unreachable=0    failed=0    skipped=2    rescued=0    ignored=2
```

File: plugins/modules/system/firewalld.py (hunk 1 above)
Comment (felixfontein) [reply plugins/modules/system/firewalld.py:91]: @maxamillion you need the version_added.

**Comment by felixfontein** [comment]
@maxamillion you marked the review comments of mine as resolved, but never pushed a change which adjusted anything in these lines. Did you forgot to push something? (I unresolved them.)

**Comment by maxamillion** [comment]
> @maxamillion you marked the review comments of mine as resolved, but never pushed a change which adjusted anything in these lines. Did you forgot to push something? (I unresolved them.)

@felixfontein no sorry, the fix I was talking about ended up being this https://github.com/ansible-collections/community.general/pull/499 ... shippable should be happy now.

**Comment by felixfontein** [comment]
@maxamillion tests passed, merged :)

**Pull request merged**